sdk/identity/identity/src/msal/msalClient.ts

```diff
@@ -0,0 +1,301 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import * as msal from "@azure/msal-node";
+
+import { AccessToken, GetTokenOptions } from "@azure/core-auth";
+import { credentialLogger, formatSuccess } from "../util/logging";
+import {
+  defaultLoggerCallback,
+  ensureValidMsalToken,
+  getAuthority,
+  getKnownAuthorities,
+  getMSALLogLevel,
+  handleMsalError,
+  publicToMsal,
+} from "./utils";
+
+import { AuthenticationRequiredError } from "../errors";
+import { IdentityClient } from "../client/identityClient";
+import { MsalNodeOptions } from "./nodeFlows/msalNodeCommon";
+import { RegionalAuthority } from "../regionalAuthority";
+import { getLogLevel } from "@azure/logger";
+import { resolveTenantId } from "../util/tenantIdUtils";
+
+/**
+ * The logger for all MsalClient instances.
+ */
+const msalLogger = credentialLogger("MsalClient");
+
+/**
+ * Interface for the MSAL (Microsoft Authentication Library) client.
+ * This client is used to interact with Microsoft's identity platform.
+ */
+export interface MsalClient {
+  /**
+   * Retrieves an access token by using a client secret.
+   *
+   * @param scopes - The scopes for which the access token is requested. These represent the resources that the application wants to access.
+   * @param clientSecret - The client secret of the application. This is a credential that the application can use to authenticate itself.
+   * @param options - Additional options that may be provided to the method.
+   * @returns An access token
+   */
+  getTokenByClientSecret(
+    scopes: string[],
+    clientSecret: string,
+    options?: GetTokenOptions,
+  ): Promise<AccessToken>;
+}
+
+/**
+ * Options for creating an instance of the MsalClient.
+ */
+export type CreateMsalClientOptions = Partial<Omit<MsalNodeOptions, "clientId" | "tenantId">>;
```

Comment:
Needs a better type. Not clear to me why `logger` is a required option in MsalNodeOptions, and I would like to avoid passing clientId and tenantId in two places

Reply:
so that we have one logger throughout the whole process that is doing the logging - that was the reason i understood